### PR TITLE
Michelle Yu assignment 3 & 4 completed

### PR DESCRIPTION
**Pull request by michellehyu**

### CODE REVIEW

**Comment by michellehyu** [comment]
Michelle Yu assignment 3 and 4 completed

File: assignments/assignment02.rb [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+median [1, 2, 3]  #=> 2
+median [1, 1, 4]  #=> 1
+
+
+# ========================================================================================
+#  Problem 3 - `pluck`
+
+# implement method `pluck` on array of hashes
+def pluck(ary, key)
+  ary.map {|item| key_ary << item[key]}
```
Comment (planetcohen) [inline assignments/assignment02.rb:92]: there's no key_ary. Map implcitly creates a new array. So your statement should look like this:
`ary.map {|item| item[key]}

File: assignments/assignment02.rb [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+# - summary:
+#   - starting balance, total deposits, total withdrawals, ending balance
+def create_transaction(date, payee, amount, type)
+  {date: date, payee: payee, amount: amount, type: type}
+end
+
+#get records based on type
+def get_records(records, type)
+  r = records.select{ |r| r[:type] == type}
+  r.sort_by!{|r| r[:date].split('/').map{|d| d.to_i} }
```
Comment (planetcohen) [inline assignments/assignment02.rb:127]: don't use the ! methods, they modify the original argument, which could have side-effects that are really hard to debug

File: assignments/assignment02.rb [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+  return r
+end
+
+#get sum of all records
+def records_sum(records)
+  sum = 0.0;
+  records.each{|r| sum+= r[:amount].to_f}
```
Comment (planetcohen) [inline assignments/assignment02.rb:134]: don't use each. Use reduce to calculate the sum.

File: assignments/assignment03.rb [hunk 4]
```diff
@@ -0,0 +1,326 @@
...
+# re-implement titleize and palindrome? as methods on String
+class String
+
+  def titleize
+    #get array of words in the s
+    words_in_s = split " "
+
+    #make a string with all the words capitalized
+    titlized_s = ""
+    words_in_s.each do |w|
```
Comment (planetcohen) [inline assignments/assignment03.rb:17]: don't use each, use map

**Comment by planetcohen** [comment]
I'm still seeing many cases where you are iterating with each to generate a new array or calculate a value. Please, please, please look at map & reduce and understand how to use those instead of iterating with each.

**Pull request closed without merging**